testcases/OpTestGSBStaticKey.py

```diff
@@ -128,86 +120,59 @@ def getPRePDisk(self):
-            # Copy the signed grub in to the PReP disk using 'dd' command
-            out = self.c.run_command("dd if=%s of=%s || echo 'no'" %
-                                     (self.grubFilename, self.prepDisk))
-            if "no" in "".join(out):
-                self.fail("RHEL: Failed to copy to PReP partition")
-        elif self.distro == 'rhel' and not enable:
-            # Nothing to do for Secure Boot disable for RHEL at OS level
-            pass
-        bfile = "/etc/sysconfig/bootloader"
-        if self.distro == 'sles' and enable:
-            # Add SECURE_BOOT="yes" at file  /etc/sysconfig/bootloader
-            cmd = "sed -i '/SECURE_BOOT=\"no\"/c\SECURE_BOOT=\"yes\"' %s" % bfile
-            self.c.run_command(cmd)
-            out = self.c.run_command("pbl --install || echo 'no' ")
-            if "no" in "".join(out):
-                self.fail("SLES: Failed to enable Secure Boot")
-        elif self.distro == 'sles' and not enable:
-            # Add SECURE_BOOT="no" at file  /etc/sysconfig/bootloader
-            cmd = "sed -i '/SECURE_BOOT=\"yes\"/c\SECURE_BOOT=\"no\"' %s" % bfile
-            self.c.run_command(cmd)
-            out = self.c.run_command("pbl --install || echo 'no'")
-            if "no" in "".join(out):
-                self.fail("SLES: Failed to disable Secure Boot")
+        if self.kernel_signature == "signed_kernel":
+            if self.distro == 'rhel' and enable:
+                # Get the PReP disk file
+                self.getPRePDisk()
+                #Proceed ahead only if the grub is signed
+                if self.grub_signature == "signed_grub":
+                    # Running grub2-install on PReP disk
+                    out = self.c.run_command("grub2-install %s" % self.prepDisk)
+                    if "Installation finished. No error reported." not in out:
+                        self.fail("RHEL: Failed to install on PReP partition")
+                    # Copy the signed grub in to the PReP disk using 'dd' command
+                    out = self.c.run_command("dd if=%s of=%s || echo 'no'" % 
+                                             (self.grub_filename, self.prepDisk))
+                    if "no" in "".join(out):
+                        self.fail("RHEL: Failed to copy to PReP partition")
+            elif self.distro == 'rhel' and not enable:
+                # Nothing to do for Secure Boot disable for RHEL at OS level
+                pass
+            bfile = "/etc/sysconfig/bootloader"
+            if self.distro == 'sles' and enable:
+                # Add SECURE_BOOT="yes" at file  /etc/sysconfig/bootloader
+                cmd = "sed -i '/SECURE_BOOT=\"no\"/c\SECURE_BOOT=\"yes\"' %s" % bfile
+                self.c.run_command(cmd)
+                out = self.c.run_command("pbl --install || echo 'no' ")
+                if "no" in "".join(out):
+                    self.fail("SLES: Failed to enable Secure Boot")
```

Comment:
Similar comments for SLES as well

Reply:
@sacsant 
- As discussed with Sastry, from SLES15-sp5 onwards, Secureboot is by-default in an enabled state at the OS level and can be checked through the "yast2 bootloader". 
- This code was written earlier, and hence here onwards is not required to be handled specifically for SLES. Hence going ahead and removing the same.